httpx/_urlparse.py

```diff
@@ -205,9 +210,15 @@ def urlparse(url: str = "", **kwargs: str | None) -> ParseResult:
             # If a component includes any ASCII control characters including \t, \r, \n,
             # then treat it as invalid.
             if any(char.isascii() and not char.isprintable() for char in value):
-                raise InvalidURL(
-                    f"Invalid non-printable ASCII character in URL component '{key}'"
+                char = [
+                    char for char in value if char.isascii() and not char.isprintable()
+                ][0]
```

Comment:
Same here. This could be a small utility function as well (it's duplicated in 2 places, I'm okay keeping it duplicated).